Feature/unread notifications

AdityaKhatri opened the pull request:
Depends on: https://github.com/the-deep/server/pull/487
Depends on https://github.com/toggle-corp/react-store/pull/734

AdityaKhatri committed:
Add mark as read button in notifications

AdityaKhatri committed:
Change status and count after status change

AdityaKhatri committed:
Add permalink for entry comments from notifications

tnagorra commented on src/components/general/Navbar/Notifica/Notifications/index.js:
```diff
@@ -93,10 +97,13 @@ const requestOptions = {
         query: ({ params: {
             dateRange,
             isPending,
+            seenStatus = 'all',
         } }) => {
+            const status = seenStatus === 'all' ? undefined : seenStatus;
```
Wut?

tnagorra commented on src/components/general/Navbar/Notifica/Notifications/items/Notification/index.js:
```diff
@@ -1,17 +1,27 @@
+
 import styles from './styles.scss';
 
+const NOTIFICATION_STATUS_UNSEEN = 'unseen';
+const NOTIFICATION_STATUS_SEEN = 'seen';
```
Why are these not used everywhere though?

tnagorra commented on src/components/general/Navbar/Notifica/Notifications/items/ProjectJoinResponse/index.js:
```diff
@@ -29,106 +32,106 @@ const emptyObject = {};
-
-        return null;
     }
+    return null;
```
It's better to return `null` before the whole component. Easier to read

AdityaKhatri committed:
Separate pending and done by group in notifications

AdityaKhatri committed:
Add allow-downloads flag in viz iframes

tnagorra committed:
Update notification locally after seen/unseen change

- Remove api use of bulk notification update
- Set schema for notifications request
- Update schema for notifications
- Remove "updateNotification" call after request accept
- Remove unused requests

tnagorra commented:
@subinasr Can you check if the latest commit works?

tnagorra commented:
@subinasr Please!
